cetmix_tower_server/security/cx_tower_plan_security.xml

```diff
@@ -6,15 +6,39 @@
         <field name="name">Tower plan: user access rule</field>
         <field name="model_id" ref="model_cx_tower_plan" />
         <field name="groups" eval="[(4, ref('cetmix_tower_server.group_user'))]" />
-        <field name="domain_force">[('access_level', '=', '1')]</field>
+        <field name="domain_force">
+            [
+            '|',
+            '&amp;',
+            ('access_level', '=', '1'),
+            ('server_ids', '=', False),
+            '&amp;',
+            ('access_level', '=', '1'),
+            ('server_ids.message_partner_ids', 'in', [user.partner_id.id])
+            ]
+
```

Comment:
Access level is always the same so we can move it out of the expressions:

```
('access_level', '=', '1'),
'|',
('server_ids', '=', False),
('server_ids.message_partner_ids', 'in', [user.partner_id.id])
```

Reply:
Thanks for the suggestion. I've updated the rules